src/runtime/statistics.py

```diff
@@ -11,54 +15,123 @@ def mean(x: list[i32]) -> f64:
 
     for i in range(k):
         sum += x[i]
-    ans: f64
-    ans = sum/k
-    return ans
+    return sum/k
+
+
+@overload
+def mean(x: list[i64]) -> f64:
+    """
+    Returns the arithmetic mean of a data sequence of numbers
+    """
+    k: i32 = len(x)
+    if k == 0:
+        return 0.0
+    sum: f64
+    sum = 0.0
+    i: i32
+
+    for i in range(k):
+        sum += x[i]
+
+    return sum/k
+
+
+@overload
+def mean(x: list[f32]) -> f64:
+    """
+    Returns the arithmetic mean of a data sequence of numbers
+    """
+    k: i32 = len(x)
+    if k == 0:
+        return 0.0
+    sum: f64
+    sum = 0.0
+    i: i32
+
+    for i in range(k):
+        sum += x[i]
+    return sum/k
```

Comment:
Btw, these are great candidates to try using the new templates. The templates are still experimental, and we still need to implement restrictions, so things will change, but the current state should be able to compile such functions.

Reply:
I agree. @ansharlubis @Oshanath May be, you can try templates on these functions and reduce the implementation size? What do you folks say?

Reply:
I think it might be a little bit early, since it will slow @ansharlubis down if we have to do any refactoring, which we will. We can however do it on a branch / Draft PR that is not merged. That would be great, just to see if it works, and to iron out any bugs, but I would not merge into master yet.

Reply:
It kind of work in an example I added in PR #921, it still uses the hard-coded restrictions though and don't do much type checks.